src/flag_gems/ops/cross_entropy_loss.py

```diff
@@ -462,11 +462,47 @@ def celoss_indice_smooth_bwd(
         tl.store(inp_grad_ptrs, inp_grad, mask=inp_mask and ignore_mask)
 
 
+@libentry()
+@triton.autotune(
+    configs=[triton.Config({"BLOCK_N": n}) for n in [64, 256, 1024]],
+    key=[
+        "N",
+    ],
+)
+@triton.jit
+def sum_and_scale(
+    inp_ptr,
+    out_ptr,
+    N,
+    ifscale: tl.constexpr,  # 0: not scale; 1: scale by scalar; 2: scale by tensor
+    BLOCK_N: tl.constexpr,
+    scale=1.0,
+):
+    mid_sum = tl.zeros(
+        [
+            BLOCK_N,
+        ],
+        dtype=tl.float32,
+    )
+    for off in range(0, N, BLOCK_N):
+        offset = off + tl.arange(0, BLOCK_N)
+        inp_ptrs = inp_ptr + offset
+        mask = offset < N
+        inp_vals = tl.load(inp_ptrs, mask=mask, other=0.0)
+        mid_sum += inp_vals
+    out_val = tl.sum(mid_sum)
+    if ifscale == 1:
+        out_val /= scale
+    elif ifscale == 2:
+        scale_val = tl.load(scale)
```

Comment:
This branching makes the type of scale ambiguous, it is either a number, or a pointer.

Reply:
replace parameter "ifscale" with "scalebyw" which means scale by weight

Reply:
and scale is a data pointer when scalebyw is true, and scalar value while false.